diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceSecuredImpl.java
index b86ec50..07c7429 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceSecuredImpl.java
@@ -32,7 +32,7 @@ import gov.hhs.fha.nhinc.common.auditlog.LogEventSecureRequestType;
 import gov.hhs.fha.nhinc.common.nhinccommon.AcknowledgementType;
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinccomponentauditrepository.AuditRepositoryManagerSecuredPortType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
@@ -82,7 +82,7 @@ public class AuditRepositoryProxyWebServiceSecuredImpl implements AuditRepositor
 
                     ServicePortDescriptor<AuditRepositoryManagerSecuredPortType> portDescriptor = new AuditRepositorySecuredServicePortDescriptor();
 
-                    CONNECTClient<AuditRepositoryManagerSecuredPortType> client = new CONNECTCXFClientFactory<AuditRepositoryManagerSecuredPortType>()
+                    CONNECTClient<AuditRepositoryManagerSecuredPortType> client = new CONNECTClientFactory<AuditRepositoryManagerSecuredPortType>()
                             .getCONNECTClientSecured(portDescriptor, url, assertion);
 
                     result = (AcknowledgementType) client.invokePort(AuditRepositoryManagerSecuredPortType.class,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceUnsecuredImpl.java
index 02383d4..5c4d9e9 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/auditrepository/nhinc/proxy/AuditRepositoryProxyWebServiceUnsecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.auditlog.LogEventRequestType;
 import gov.hhs.fha.nhinc.common.nhinccommon.AcknowledgementType;
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinccomponentauditrepository.AuditRepositoryManagerPortType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
@@ -74,7 +74,7 @@ public class AuditRepositoryProxyWebServiceUnsecuredImpl implements AuditReposit
 
                     ServicePortDescriptor<AuditRepositoryManagerPortType> portDescriptor = new AuditRepositoryUnsecuredServicePortDescriptor();
 
-                    CONNECTClient<AuditRepositoryManagerPortType> client = new CONNECTCXFClientFactory<AuditRepositoryManagerPortType>()
+                    CONNECTClient<AuditRepositoryManagerPortType> client = new CONNECTClientFactory<AuditRepositoryManagerPortType>()
                             .getCONNECTClientUnsecured(portDescriptor, url, assertion);
                     
                     result = (AcknowledgementType) client.invokePort(AuditRepositoryManagerPortType.class, "logEvent", request);
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/request/proxy/AdapterDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/request/proxy/AdapterDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
index 5e0f335..36a0c8c 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/request/proxy/AdapterDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/request/proxy/AdapterDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
@@ -36,7 +36,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.common.nhinccommonadapter.AdapterProvideAndRegisterDocumentSetSecuredRequestType;
 import gov.hhs.fha.nhinc.docsubmission.adapter.deferred.request.proxy.service.AdapterDocSubmissionDeferredRequestSecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NullChecker;
@@ -69,7 +69,7 @@ public class AdapterDocSubmissionDeferredRequestProxyWebServiceSecuredImpl imple
 
     protected CONNECTClient<AdapterXDRRequestSecuredPortType> getCONNECTClientSecured(
             ServicePortDescriptor<AdapterXDRRequestSecuredPortType> portDescriptor, String url, AssertionType assertion) {
-        return new CONNECTCXFClientFactory<AdapterXDRRequestSecuredPortType>().getCONNECTClientSecured(portDescriptor,
+        return new CONNECTClientFactory<AdapterXDRRequestSecuredPortType>().getCONNECTClientSecured(portDescriptor,
                 url, assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/request/proxy/AdapterDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/request/proxy/AdapterDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
index a9fbbd2..8cb269b 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/request/proxy/AdapterDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/request/proxy/AdapterDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
@@ -36,7 +36,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.common.nhinccommonadapter.AdapterProvideAndRegisterDocumentSetRequestType;
 import gov.hhs.fha.nhinc.docsubmission.adapter.deferred.request.proxy.service.AdapterDocSubmissionDeferredRequestUnsecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NullChecker;
@@ -67,7 +67,7 @@ public class AdapterDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl imp
 
     protected CONNECTClient<AdapterXDRRequestPortType> getCONNECTClientUnsecured(
             ServicePortDescriptor<AdapterXDRRequestPortType> portDescriptor, String url, AssertionType assertion) {
-        return new CONNECTCXFClientFactory<AdapterXDRRequestPortType>().getCONNECTClientUnsecured(portDescriptor, url,
+        return new CONNECTClientFactory<AdapterXDRRequestPortType>().getCONNECTClientUnsecured(portDescriptor, url,
                 assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/response/proxy/AdapterDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/response/proxy/AdapterDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
index 1c3222a..562d04e 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/response/proxy/AdapterDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/response/proxy/AdapterDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
@@ -30,7 +30,7 @@ import gov.hhs.fha.nhinc.adapterxdrresponsesecured.AdapterXDRResponseSecuredPort
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.docsubmission.adapter.deferred.response.proxy.service.AdapterDocSubmissionDeferredResponseSecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NullChecker;
@@ -67,7 +67,7 @@ public class AdapterDocSubmissionDeferredResponseProxyWebServiceSecuredImpl impl
     protected CONNECTClient<AdapterXDRResponseSecuredPortType> getCONNECTClientSecured(
             ServicePortDescriptor<AdapterXDRResponseSecuredPortType> portDescriptor, String url, AssertionType assertion) {
         
-        return new CONNECTCXFClientFactory<AdapterXDRResponseSecuredPortType>().getCONNECTClientSecured(portDescriptor,
+        return new CONNECTClientFactory<AdapterXDRResponseSecuredPortType>().getCONNECTClientSecured(portDescriptor,
                 url, assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/response/proxy/AdapterDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/response/proxy/AdapterDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
index 94214ed..43f488b 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/response/proxy/AdapterDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/deferred/response/proxy/AdapterDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.common.nhinccommonadapter.AdapterRegistryResponseType;
 import gov.hhs.fha.nhinc.docsubmission.adapter.deferred.response.proxy.service.AdapterDocSubmissionDeferredResponseUnsecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NullChecker;
@@ -67,7 +67,7 @@ public class AdapterDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl im
     protected CONNECTClient<AdapterXDRResponsePortType> getCONNECTClientUnsecured(
             ServicePortDescriptor<AdapterXDRResponsePortType> portDescriptor, String url, AssertionType assertion) {
         
-        return new CONNECTCXFClientFactory<AdapterXDRResponsePortType>().getCONNECTClientUnsecured(portDescriptor, url, assertion);
+        return new CONNECTClientFactory<AdapterXDRResponsePortType>().getCONNECTClientUnsecured(portDescriptor, url, assertion);
     } 
 
     public XDRAcknowledgementType provideAndRegisterDocumentSetBResponse(RegistryResponseType regResponse,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/proxy/AdapterDocSubmissionProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/proxy/AdapterDocSubmissionProxyWebServiceSecuredImpl.java
index f18689b..dab5089 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/proxy/AdapterDocSubmissionProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/proxy/AdapterDocSubmissionProxyWebServiceSecuredImpl.java
@@ -30,7 +30,7 @@ import gov.hhs.fha.nhinc.adapterxdrsecured.AdapterXDRSecuredPortType;
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.docsubmission.adapter.proxy.service.AdapterDocSubmissionSecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NullChecker;
@@ -74,7 +74,7 @@ public class AdapterDocSubmissionProxyWebServiceSecuredImpl implements AdapterDo
 
                 ServicePortDescriptor<AdapterXDRSecuredPortType> portDescriptor = new AdapterDocSubmissionSecuredServicePortDescriptor();
 
-                CONNECTClient<AdapterXDRSecuredPortType> client = new CONNECTCXFClientFactory<AdapterXDRSecuredPortType>()
+                CONNECTClient<AdapterXDRSecuredPortType> client = new CONNECTClientFactory<AdapterXDRSecuredPortType>()
                         .getCONNECTClientSecured(portDescriptor, url, assertion);
 
                 response = (RegistryResponseType) client.invokePort(AdapterXDRSecuredPortType.class,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/proxy/AdapterDocSubmissionProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/proxy/AdapterDocSubmissionProxyWebServiceUnsecuredImpl.java
index 237cc4a..4dd22f3 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/proxy/AdapterDocSubmissionProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/adapter/proxy/AdapterDocSubmissionProxyWebServiceUnsecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.common.nhinccommonadapter.AdapterProvideAndRegisterDocumentSetRequestType;
 import gov.hhs.fha.nhinc.docsubmission.adapter.proxy.service.AdapterDocSubmissionUnsecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NullChecker;
@@ -78,7 +78,7 @@ public class AdapterDocSubmissionProxyWebServiceUnsecuredImpl implements Adapter
                 
                 ServicePortDescriptor<AdapterXDRPortType> portDescriptor = new AdapterDocSubmissionUnsecuredServicePortDescriptor();
                 
-                CONNECTClient<AdapterXDRPortType> client = new CONNECTCXFClientFactory<AdapterXDRPortType>()
+                CONNECTClient<AdapterXDRPortType> client = new CONNECTClientFactory<AdapterXDRPortType>()
                         .getCONNECTClientUnsecured(portDescriptor, url, assertion);
                 
                 response = (RegistryResponseType) client.invokePort(AdapterXDRPortType.class,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/proxy/EntityDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/proxy/EntityDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
index 88fa617..bc2ff03 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/proxy/EntityDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/proxy/EntityDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
@@ -38,7 +38,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.UrlInfoType;
 import gov.hhs.fha.nhinc.common.nhinccommonentity.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType;
 import gov.hhs.fha.nhinc.docsubmission.entity.deferred.request.proxy.service.EntityDocSubmissionDeferredRequestSecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhincentityxdrsecured.async.request.EntityXDRSecuredAsyncRequestPortType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
@@ -71,7 +71,7 @@ public class EntityDocSubmissionDeferredRequestProxyWebServiceSecuredImpl implem
             ServicePortDescriptor<EntityXDRSecuredAsyncRequestPortType> portDescriptor, String url,
             AssertionType assertion) {
 
-        return new CONNECTCXFClientFactory<EntityXDRSecuredAsyncRequestPortType>().getCONNECTClientSecured(portDescriptor,
+        return new CONNECTClientFactory<EntityXDRSecuredAsyncRequestPortType>().getCONNECTClientSecured(portDescriptor,
                 url, assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/proxy/EntityDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/proxy/EntityDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
index 994e042..8d448bd 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/proxy/EntityDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/request/proxy/EntityDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
@@ -38,7 +38,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.UrlInfoType;
 import gov.hhs.fha.nhinc.common.nhinccommonentity.RespondingGatewayProvideAndRegisterDocumentSetRequestType;
 import gov.hhs.fha.nhinc.docsubmission.entity.deferred.request.proxy.service.EntityDocSubmissionDeferredRequestServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhincentityxdr.async.request.EntityXDRAsyncRequestPortType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
@@ -70,7 +70,7 @@ public class EntityDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl impl
     protected CONNECTClient<EntityXDRAsyncRequestPortType> getCONNECTClientUnsecured(
             ServicePortDescriptor<EntityXDRAsyncRequestPortType> portDescriptor, String url, AssertionType assertion) {
 
-        return new CONNECTCXFClientFactory<EntityXDRAsyncRequestPortType>().getCONNECTClientUnsecured(portDescriptor, url,
+        return new CONNECTClientFactory<EntityXDRAsyncRequestPortType>().getCONNECTClientUnsecured(portDescriptor, url,
                 assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/proxy/EntityDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/proxy/EntityDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
index 9542c99..576991d 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/proxy/EntityDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/proxy/EntityDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetCommunitiesType;
 import gov.hhs.fha.nhinc.common.nhinccommonentity.RespondingGatewayProvideAndRegisterDocumentSetSecuredResponseRequestType;
 import gov.hhs.fha.nhinc.docsubmission.entity.deferred.response.proxy.service.EntityDocSubmissionDeferredResponseSecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhincentityxdrsecured.async.response.EntityXDRSecuredAsyncResponsePortType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
@@ -83,7 +83,7 @@ public class EntityDocSubmissionDeferredResponseProxyWebServiceSecuredImpl imple
 
                 ServicePortDescriptor<EntityXDRSecuredAsyncResponsePortType> portDescriptor = new EntityDocSubmissionDeferredResponseSecuredServicePortDescriptor();
 
-                CONNECTClient<EntityXDRSecuredAsyncResponsePortType> client = CONNECTCXFClientFactory.getInstance()
+                CONNECTClient<EntityXDRSecuredAsyncResponsePortType> client = new CONNECTClientFactory<EntityXDRSecuredAsyncResponsePortType>()
                         .getCONNECTClientSecured(portDescriptor, url, assertion);
 
                 response = (XDRAcknowledgementType) client.invokePort(EntityXDRSecuredAsyncResponsePortType.class,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/proxy/EntityDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/proxy/EntityDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
index 618381c..15a6da7 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/proxy/EntityDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/deferred/response/proxy/EntityDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
@@ -30,8 +30,8 @@ import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetCommunitiesType;
 import gov.hhs.fha.nhinc.common.nhinccommonentity.RespondingGatewayProvideAndRegisterDocumentSetResponseRequestType;
 import gov.hhs.fha.nhinc.docsubmission.entity.deferred.response.proxy.service.EntityDocSubmissionDeferredResponseUnsecuredServicePortDescriptor;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhincentityxdr.async.response.EntityXDRAsyncResponsePortType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
@@ -86,7 +86,7 @@ public class EntityDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl imp
 
                 ServicePortDescriptor<EntityXDRAsyncResponsePortType> portDescriptor = new EntityDocSubmissionDeferredResponseUnsecuredServicePortDescriptor();
 
-                CONNECTClient<EntityXDRAsyncResponsePortType> client = CONNECTCXFClientFactory.getInstance()
+                CONNECTClient<EntityXDRAsyncResponsePortType> client = new CONNECTClientFactory<EntityXDRAsyncResponsePortType>()
                         .getCONNECTClientUnsecured(portDescriptor, url, assertion);
 
                 response = (XDRAcknowledgementType) client.invokePort(EntityXDRAsyncResponsePortType.class,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceSecuredImpl.java
index aaee4b8..9a8d8b1 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceSecuredImpl.java
@@ -32,7 +32,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.UrlInfoType;
 import gov.hhs.fha.nhinc.common.nhinccommonentity.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType;
 import gov.hhs.fha.nhinc.docsubmission.entity.proxy.description.EntityDocSubmissionSecureda0ServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhincentityxdrsecured.EntityXDRSecuredPortType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
@@ -102,6 +102,6 @@ public class EntityDocSubmissionProxyWebServiceSecuredImpl implements EntityDocS
      */
     protected CONNECTClient<EntityXDRSecuredPortType> getCONNECTClient(
             ServicePortDescriptor<EntityXDRSecuredPortType> portDescriptor, String url, AssertionType assertion) {
-        return CONNECTCXFClientFactory.getInstance().getCONNECTClientSecured(portDescriptor, url, assertion);
+        return new CONNECTClientFactory<EntityXDRSecuredPortType>().getCONNECTClientSecured(portDescriptor, url, assertion);
     }
 }
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceUnsecuredImpl.java
index a6693ac..fc45d8f 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/entity/proxy/EntityDocSubmissionProxyWebServiceUnsecuredImpl.java
@@ -32,7 +32,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.UrlInfoType;
 import gov.hhs.fha.nhinc.common.nhinccommonentity.RespondingGatewayProvideAndRegisterDocumentSetRequestType;
 import gov.hhs.fha.nhinc.docsubmission.entity.proxy.description.EntityDocSubmissiona0ServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhincentityxdr.EntityXDRPortType;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
@@ -101,6 +101,6 @@ public class EntityDocSubmissionProxyWebServiceUnsecuredImpl implements EntityDo
      */
     protected CONNECTClient<EntityXDRPortType> getCONNECTClient(ServicePortDescriptor<EntityXDRPortType> portDescriptor,
             String url, AssertionType assertion) {
-        return CONNECTCXFClientFactory.getInstance().getCONNECTClientUnsecured(portDescriptor, url, assertion);
+        return new CONNECTClientFactory<EntityXDRPortType>().getCONNECTClientUnsecured(portDescriptor, url, assertion);
     }
 }
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/request/proxy20/NhinDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/request/proxy20/NhinDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
index 30bc6e8..c67819d 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/request/proxy20/NhinDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/request/proxy20/NhinDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
@@ -30,7 +30,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.docsubmission.nhin.deferred.request.proxy20.service.NhinDocSubmissionDeferredRequestServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants.GATEWAY_API_LEVEL;
@@ -68,7 +68,7 @@ public class NhinDocSubmissionDeferredRequestProxyWebServiceSecuredImpl implemen
     protected CONNECTClient<XDRDeferredRequest20PortType> getCONNECTClientSecured(
             ServicePortDescriptor<XDRDeferredRequest20PortType> portDescriptor, String url, AssertionType assertion) {
 
-        return CONNECTCXFClientFactory.getInstance().getCONNECTClientSecured(portDescriptor, url,
+        return new CONNECTClientFactory<XDRDeferredRequest20PortType>().getCONNECTClientSecured(portDescriptor, url,
                 assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/response/proxy11/NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/response/proxy11/NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
index b1a19d0..cc73963 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/response/proxy11/NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/response/proxy11/NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
@@ -75,7 +75,7 @@ public class NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl impleme
     protected CONNECTClient<XDRDeferredResponsePortType> getCONNECTClientSecured(
             ServicePortDescriptor<XDRDeferredResponsePortType> portDescriptor, String url, AssertionType assertion) {
 
-        return CONNECTClientFactory.getInstance().getCONNECTClientSecured(portDescriptor, url,
+        return new CONNECTClientFactory<XDRDeferredResponsePortType>().getCONNECTClientSecured(portDescriptor, url,
                 assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/response/proxy20/NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/response/proxy20/NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
index 8129af2..8a086f8 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/response/proxy20/NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/deferred/response/proxy20/NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
@@ -30,7 +30,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.docsubmission.nhin.deferred.response.proxy20.service.NhinDocSubmissionDeferredResponseServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants.GATEWAY_API_LEVEL;
@@ -70,7 +70,7 @@ public class NhinDocSubmissionDeferredResponseProxyWebServiceSecuredImpl impleme
     protected CONNECTClient<XDRDeferredResponse20PortType> getCONNECTClientSecured(
             ServicePortDescriptor<XDRDeferredResponse20PortType> portDescriptor, String url, AssertionType assertion) {
         
-        return CONNECTCXFClientFactory.getInstance().getCONNECTClientSecured(portDescriptor,
+        return new CONNECTClientFactory<XDRDeferredResponse20PortType>().getCONNECTClientSecured(portDescriptor,
                 url, assertion);
     }
     
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/proxy/NhinDocSubmissionProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/proxy/NhinDocSubmissionProxyWebServiceSecuredImpl.java
index b3fccca..2b63e79 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/proxy/NhinDocSubmissionProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/nhin/proxy/NhinDocSubmissionProxyWebServiceSecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.docsubmission.nhin.proxy.service.NhinDocSubmission20ServicePortDescriptor;
 import gov.hhs.fha.nhinc.docsubmission.nhin.proxy.service.NhinDocSubmissionServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.webserviceproxy.WebServiceProxyHelper;
@@ -81,7 +81,7 @@ public class NhinDocSubmissionProxyWebServiceSecuredImpl implements NhinDocSubmi
 
             ServicePortDescriptor<DocumentRepositoryXDRPortType> portDescriptor = getServicePortDescriptor(apiLevel);
 
-            CONNECTClient<DocumentRepositoryXDRPortType> client = CONNECTCXFClientFactory.getInstance()
+            CONNECTClient<DocumentRepositoryXDRPortType> client = new CONNECTClientFactory<DocumentRepositoryXDRPortType>()
                     .getCONNECTClientSecured(portDescriptor, url, assertion);
 
             response = (RegistryResponseType) client.invokePort(DocumentRepositoryXDRPortType.class,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/request/proxy/PassthruDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/request/proxy/PassthruDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
index ff1eafa..bbaa8cf 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/request/proxy/PassthruDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/request/proxy/PassthruDocSubmissionDeferredRequestProxyWebServiceSecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType;
 import gov.hhs.fha.nhinc.docsubmission.passthru.deferred.request.proxy.service.PassthruDocSubmissionDeferredRequestSecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhincproxyxdrsecured.async.request.ProxyXDRSecuredAsyncRequestPortType;
@@ -70,7 +70,7 @@ public class PassthruDocSubmissionDeferredRequestProxyWebServiceSecuredImpl impl
             ServicePortDescriptor<ProxyXDRSecuredAsyncRequestPortType> portDescriptor, String url,
             AssertionType assertion) {
 
-        return CONNECTCXFClientFactory.getInstance().getCONNECTClientSecured(portDescriptor,
+        return new CONNECTClientFactory<ProxyXDRSecuredAsyncRequestPortType>().getCONNECTClientSecured(portDescriptor,
                 url, assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/request/proxy/PassthruDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/request/proxy/PassthruDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
index e15056f..3ec61b2 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/request/proxy/PassthruDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/request/proxy/PassthruDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetRequestType;
 import gov.hhs.fha.nhinc.docsubmission.passthru.deferred.request.proxy.service.PassthruDocSubmissionDeferredRequestUnsecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhincproxyxdr.async.request.ProxyXDRAsyncRequestPortType;
@@ -68,7 +68,7 @@ public class PassthruDocSubmissionDeferredRequestProxyWebServiceUnsecuredImpl im
     protected CONNECTClient<ProxyXDRAsyncRequestPortType> getCONNECTClientUnsecured(
             ServicePortDescriptor<ProxyXDRAsyncRequestPortType> portDescriptor, String url, AssertionType assertion) {
 
-        return CONNECTCXFClientFactory.getInstance().getCONNECTClientUnsecured(portDescriptor, url,
+        return new CONNECTClientFactory<ProxyXDRAsyncRequestPortType>().getCONNECTClientUnsecured(portDescriptor, url,
                 assertion);
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/response/proxy/PassthruDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/response/proxy/PassthruDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
index c3f2de0..590a61c 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/response/proxy/PassthruDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/response/proxy/PassthruDocSubmissionDeferredResponseProxyWebServiceSecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetSecuredResponseRequestType;
 import gov.hhs.fha.nhinc.docsubmission.passthru.deferred.response.proxy.service.PassthruDocSubmissionDeferredResponseSecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhincproxyxdrsecured.async.response.ProxyXDRSecuredAsyncResponsePortType;
@@ -68,7 +68,7 @@ public class PassthruDocSubmissionDeferredResponseProxyWebServiceSecuredImpl imp
     protected CONNECTClient<ProxyXDRSecuredAsyncResponsePortType> getCONNECTClientSecured(
             ServicePortDescriptor<ProxyXDRSecuredAsyncResponsePortType> portDescriptor, String url, AssertionType assertion) {
         
-    return CONNECTCXFClientFactory.getInstance().getCONNECTClientSecured(portDescriptor, url, assertion);
+    return new CONNECTClientFactory<ProxyXDRSecuredAsyncResponsePortType>().getCONNECTClientSecured(portDescriptor, url, assertion);
 }
     
     public XDRAcknowledgementType provideAndRegisterDocumentSetBResponse(RegistryResponseType request,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/response/proxy/PassthruDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/response/proxy/PassthruDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
index 34bff1b..05c7f16 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/response/proxy/PassthruDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/deferred/response/proxy/PassthruDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetResponseRequestType;
 import gov.hhs.fha.nhinc.docsubmission.passthru.deferred.response.proxy.service.PassthruDocSubmissionDeferredResponseUnsecuredServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhincproxyxdr.async.response.ProxyXDRAsyncResponsePortType;
@@ -67,7 +67,7 @@ public class PassthruDocSubmissionDeferredResponseProxyWebServiceUnsecuredImpl i
     protected CONNECTClient<ProxyXDRAsyncResponsePortType> getCONNECTClientUnsecured(
             ServicePortDescriptor<ProxyXDRAsyncResponsePortType> portDescriptor, String url, AssertionType assertion) {
         
-    return CONNECTCXFClientFactory.getInstance().getCONNECTClientUnsecured(portDescriptor, url, assertion);
+    return new CONNECTClientFactory<ProxyXDRAsyncResponsePortType>().getCONNECTClientUnsecured(portDescriptor, url, assertion);
 }
 
     public XDRAcknowledgementType provideAndRegisterDocumentSetBResponse(RegistryResponseType request,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/proxy/PassthruDocSubmissionProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/proxy/PassthruDocSubmissionProxyWebServiceSecuredImpl.java
index e47b82a..60c89a4 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/proxy/PassthruDocSubmissionProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/proxy/PassthruDocSubmissionProxyWebServiceSecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType;
 import gov.hhs.fha.nhinc.docsubmission.passthru.proxy.description.PassthruDocSubmissionSecureda0ServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhincproxyxdrsecured.ProxyXDRSecuredPortType;
@@ -77,7 +77,7 @@ public class PassthruDocSubmissionProxyWebServiceSecuredImpl implements Passthru
             
             ServicePortDescriptor<ProxyXDRSecuredPortType> portDescriptor = getServicePortDescriptor(NhincConstants.ADAPTER_API_LEVEL.LEVEL_a0);
 
-            CONNECTClient<ProxyXDRSecuredPortType> client = CONNECTCXFClientFactory.getInstance()
+            CONNECTClient<ProxyXDRSecuredPortType> client = new CONNECTClientFactory<ProxyXDRSecuredPortType>()
                     .getCONNECTClientSecured(portDescriptor, url, assertion);
 
             RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType request = new RespondingGatewayProvideAndRegisterDocumentSetSecuredRequestType();
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/proxy/PassthruDocSubmissionProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/proxy/PassthruDocSubmissionProxyWebServiceUnsecuredImpl.java
index 7f00ee9..2a8fc57 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/proxy/PassthruDocSubmissionProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/docsubmission/passthru/proxy/PassthruDocSubmissionProxyWebServiceUnsecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
 import gov.hhs.fha.nhinc.common.nhinccommonproxy.RespondingGatewayProvideAndRegisterDocumentSetRequestType;
 import gov.hhs.fha.nhinc.docsubmission.passthru.proxy.description.PassthruDocSubmissiona0ServicePortDescriptor;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhincproxyxdr.ProxyXDRPortType;
@@ -77,7 +77,7 @@ public class PassthruDocSubmissionProxyWebServiceUnsecuredImpl implements Passth
 
             ServicePortDescriptor<ProxyXDRPortType> portDescriptor = getServicePortDescriptor(NhincConstants.ADAPTER_API_LEVEL.LEVEL_a0);
 
-            CONNECTClient<ProxyXDRPortType> client = CONNECTCXFClientFactory.getInstance()
+            CONNECTClient<ProxyXDRPortType> client = new CONNECTClientFactory<ProxyXDRPortType>()
                     .getCONNECTClientUnsecured(portDescriptor, url, assertion);
 
             RespondingGatewayProvideAndRegisterDocumentSetRequestType request = new RespondingGatewayProvideAndRegisterDocumentSetRequestType();
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClient.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClient.java
deleted file mode 100644
index 86e8cab..0000000
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClient.java
+++ /dev/null
@@ -1,39 +0,0 @@
-/**
- * 
- */
-package gov.hhs.fha.nhinc.messaging.client;
-
-import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
-import gov.hhs.fha.nhinc.messaging.service.ServiceEndpoint;
-import gov.hhs.fha.nhinc.messaging.service.decorator.cxf.WsAddressingServiceEndpointDecorator;
-import gov.hhs.fha.nhinc.messaging.service.port.CXFServicePortBuilder;
-import gov.hhs.fha.nhinc.messaging.service.port.ServicePortBuilder;
-import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
-
-/**
- * @author bhumphrey
- *
- */
-public abstract class CONNECTCXFClient<T> extends CONNECTClient<T> {
-
-    protected ServiceEndpoint<T> serviceEndpoint = null;
-
-    CONNECTCXFClient(ServicePortDescriptor<T> portDescriptor, String url, AssertionType assertion) {
-        super();
-
-        String wsAddressingAction = portDescriptor.getWSAddressingAction();
-
-        ServicePortBuilder<T> portBuilder = new CXFServicePortBuilder<T>(portDescriptor);
-        
-        serviceEndpoint = super.configureBasePort(portBuilder.createPort(), url);
-      
-        // CXF specific decorator configuration
-        serviceEndpoint = new WsAddressingServiceEndpointDecorator<T>(serviceEndpoint, url, wsAddressingAction,
-                assertion);
-    }
-
-    public T getPort() {
-        return serviceEndpoint.getPort();
-    }
-    
-}
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientFactory.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientFactory.java
deleted file mode 100644
index 15a38bc..0000000
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientFactory.java
+++ /dev/null
@@ -1,51 +0,0 @@
-/*
- * Copyright (c) 2012, United States Government, as represented by the Secretary of Health and Human Services. 
- * All rights reserved. 
- *
- * Redistribution and use in source and binary forms, with or without 
- * modification, are permitted provided that the following conditions are met: 
- *     * Redistributions of source code must retain the above 
- *       copyright notice, this list of conditions and the following disclaimer. 
- *     * Redistributions in binary form must reproduce the above copyright 
- *       notice, this list of conditions and the following disclaimer in the documentation 
- *       and/or other materials provided with the distribution. 
- *     * Neither the name of the United States Government nor the 
- *       names of its contributors may be used to endorse or promote products 
- *       derived from this software without specific prior written permission. 
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND 
- * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED 
- * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
- * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY 
- * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES 
- * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
- * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND 
- * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
- * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
- */
-
-package gov.hhs.fha.nhinc.messaging.client;
-
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientUnsecured;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
-import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
-
-/**
- * @author akong
- * 
- */
-public class CONNECTCXFClientFactory<T> {
-
-    public CONNECTClient<T> getCONNECTClientSecured(ServicePortDescriptor<T> portDescriptor, String url,
-            AssertionType assertion) {
-        return new CONNECTCXFClientSecured<T>(portDescriptor, url, assertion);
-    }
-
-    public CONNECTClient<T> getCONNECTClientUnsecured(ServicePortDescriptor<T> portDescriptor, String url,
-            AssertionType assertion) {
-        return new CONNECTCXFClientUnsecured<T>(portDescriptor, url, assertion);
-    }
-
-}
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientSecured.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientSecured.java
index 0d8e39f..6a86a1f 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientSecured.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientSecured.java
@@ -27,40 +27,41 @@
 
 package gov.hhs.fha.nhinc.messaging.client;
 
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.messaging.service.ServiceEndpoint;
 import gov.hhs.fha.nhinc.messaging.service.decorator.SAMLServiceEndpointDecorator;
 import gov.hhs.fha.nhinc.messaging.service.decorator.cxf.SecurityOutInterceptorServiceEndpointDecorator;
 import gov.hhs.fha.nhinc.messaging.service.decorator.cxf.TLSClientServiceEndpointDecorator;
-<<<<<<< HEAD
 import gov.hhs.fha.nhinc.messaging.service.decorator.cxf.WsAddressingServiceEndpointDecorator;
 import gov.hhs.fha.nhinc.messaging.service.port.CXFServicePortBuilderWithAddressing;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortBuilder;
-=======
->>>>>>> a2d1b9b... refactor to change how CONNECTCXFClient works
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 
 /**
  * @author akong
  * 
  */
-public class CONNECTCXFClientSecured<T> extends CONNECTCXFClient<T> {
+public class CONNECTCXFClientSecured<T> extends CONNECTClient<T> {
+
+    private ServiceEndpoint<T> serviceEndpoint = null;
 
-   
     CONNECTCXFClientSecured(ServicePortDescriptor<T> portDescriptor, String url, AssertionType assertion) {
-        super(portDescriptor, url, assertion);
+        super();
+
+        String wsAddressingAction = portDescriptor.getWSAddressingAction();
 
-<<<<<<< HEAD
         ServicePortBuilder<T> portBuilder = new CXFServicePortBuilderWithAddressing<T>(portDescriptor);
         
         serviceEndpoint = super.configureBasePort(portBuilder.createPort(), url);
-=======
->>>>>>> a2d1b9b... refactor to change how CONNECTCXFClient works
         serviceEndpoint = new SAMLServiceEndpointDecorator<T>(serviceEndpoint, assertion);
 
+        // CXF specific decorator configuration
+        serviceEndpoint = new WsAddressingServiceEndpointDecorator<T>(serviceEndpoint, url, wsAddressingAction,
+                assertion);
         serviceEndpoint = new TLSClientServiceEndpointDecorator<T>(serviceEndpoint);
         serviceEndpoint = new SecurityOutInterceptorServiceEndpointDecorator<T>(serviceEndpoint);
 
-        
         serviceEndpoint.configure();
     }
 
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientUnsecured.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientUnsecured.java
index b67e5d5..d4b9afc 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientUnsecured.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientUnsecured.java
@@ -38,12 +38,22 @@ import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
  * @author akong
  * 
  */
-public class CONNECTCXFClientUnsecured<T>  extends CONNECTCXFClient<T>  {
+public class CONNECTCXFClientUnsecured<T> extends CONNECTClient<T> {
 
     private ServiceEndpoint<T> serviceEndpoint = null;
 
     CONNECTCXFClientUnsecured(ServicePortDescriptor<T> portDescriptor, String url, AssertionType assertion) {
-        super(portDescriptor, url, assertion );
+        super();
+
+        String wsAddressingAction = portDescriptor.getWSAddressingAction();
+
+        ServicePortBuilder<T> portBuilder = new CXFServicePortBuilder<T>(portDescriptor);
+
+        serviceEndpoint = super.configureBasePort(portBuilder.createPort(), url);
+
+        // CXF specific decorator configuration
+        serviceEndpoint = new WsAddressingServiceEndpointDecorator<T>(serviceEndpoint, url, wsAddressingAction,
+                assertion);
 
         serviceEndpoint.configure();
     }
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTClientFactory.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTClientFactory.java
index a14905f..39d4a48 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTClientFactory.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTClientFactory.java
@@ -1,19 +1,51 @@
-/**
- * 
+/*
+ * Copyright (c) 2012, United States Government, as represented by the Secretary of Health and Human Services. 
+ * All rights reserved. 
+ *
+ * Redistribution and use in source and binary forms, with or without 
+ * modification, are permitted provided that the following conditions are met: 
+ *     * Redistributions of source code must retain the above 
+ *       copyright notice, this list of conditions and the following disclaimer. 
+ *     * Redistributions in binary form must reproduce the above copyright 
+ *       notice, this list of conditions and the following disclaimer in the documentation 
+ *       and/or other materials provided with the distribution. 
+ *     * Neither the name of the United States Government nor the 
+ *       names of its contributors may be used to endorse or promote products 
+ *       derived from this software without specific prior written permission. 
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND 
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED 
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY 
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES 
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND 
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
  */
+
 package gov.hhs.fha.nhinc.messaging.client;
 
+import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientUnsecured;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 
 /**
- * @author bhumphrey
+ * @author akong
  * 
  */
-public interface CONNECTClientFactory<T> {
+public class CONNECTClientFactory<T> {
+
     public CONNECTClient<T> getCONNECTClientSecured(ServicePortDescriptor<T> portDescriptor, String url,
-            AssertionType assertion);
+            AssertionType assertion) {
+        return new CONNECTCXFClientSecured<T>(portDescriptor, url, assertion);
+    }
 
     public CONNECTClient<T> getCONNECTClientUnsecured(ServicePortDescriptor<T> portDescriptor, String url,
-            AssertionType assertion);
+            AssertionType assertion) {
+        return new CONNECTCXFClientUnsecured<T>(portDescriptor, url, assertion);
+    }
+
 }
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTMetroClientSecured.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTMetroClientSecured.java
index cbdf6e8..a8c017b 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTMetroClientSecured.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTMetroClientSecured.java
@@ -28,7 +28,6 @@
 package gov.hhs.fha.nhinc.messaging.client;
 
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
-import gov.hhs.fha.nhinc.messaging.service.BaseServiceEndpoint;
 import gov.hhs.fha.nhinc.messaging.service.ServiceEndpoint;
 import gov.hhs.fha.nhinc.messaging.service.decorator.SAMLServiceEndpointDecorator;
 import gov.hhs.fha.nhinc.messaging.service.decorator.metro.WsAddressingServiceEndpointDecorator;
@@ -51,7 +50,7 @@ public class CONNECTMetroClientSecured<T> extends CONNECTClient<T> {
 
         ServicePortBuilder<T> portBuilder = new MetroServicePortBuilder<T>(portDescriptor);
 
-        ServiceEndpoint<T> serviceEndpoint = new BaseServiceEndpoint<T>(portBuilder.createPort());
+        serviceEndpoint = super.configureBasePort(portBuilder.createPort(), url);
         serviceEndpoint = new SAMLServiceEndpointDecorator<T>(serviceEndpoint, assertion);
         
         // Metro specific decorator configuration
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTMetroClientUnsecured.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTMetroClientUnsecured.java
index c5acdde..c66eb20 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTMetroClientUnsecured.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTMetroClientUnsecured.java
@@ -29,7 +29,6 @@ package gov.hhs.fha.nhinc.messaging.client;
 
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
 import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
-import gov.hhs.fha.nhinc.messaging.service.BaseServiceEndpoint;
 import gov.hhs.fha.nhinc.messaging.service.ServiceEndpoint;
 import gov.hhs.fha.nhinc.messaging.service.decorator.metro.WsAddressingServiceEndpointDecorator;
 import gov.hhs.fha.nhinc.messaging.service.port.MetroServicePortBuilder;
@@ -51,8 +50,8 @@ public class CONNECTMetroClientUnsecured<T> extends CONNECTClient<T> {
         
         ServicePortBuilder<T> portBuilder = new MetroServicePortBuilder<T>(portDescriptor);
 
-        ServiceEndpoint<T> serviceEndpoint = new BaseServiceEndpoint<T>(portBuilder.createPort());
-        
+        serviceEndpoint = super.configureBasePort(portBuilder.createPort(), url);
+
         // Metro specific decorator configuration
         serviceEndpoint = new WsAddressingServiceEndpointDecorator<T>(serviceEndpoint, url, wsAddressingAction,
                 assertion);
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/proxy/PolicyEngineProxyWebServiceSecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/proxy/PolicyEngineProxyWebServiceSecuredImpl.java
index 929948b..d3a2c9a 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/proxy/PolicyEngineProxyWebServiceSecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/proxy/PolicyEngineProxyWebServiceSecuredImpl.java
@@ -32,7 +32,7 @@ import gov.hhs.fha.nhinc.common.nhinccommonadapter.CheckPolicyRequestSecuredType
 import gov.hhs.fha.nhinc.common.nhinccommonadapter.CheckPolicyRequestType;
 import gov.hhs.fha.nhinc.common.nhinccommonadapter.CheckPolicyResponseType;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NullChecker;
@@ -79,7 +79,7 @@ public class PolicyEngineProxyWebServiceSecuredImpl implements PolicyEngineProxy
 
                 ServicePortDescriptor<AdapterPolicyEngineSecuredPortType> portDescriptor = new PolicyEngineSecuredServicePortDescriptor();
 
-                CONNECTClient<AdapterPolicyEngineSecuredPortType> client = CONNECTCXFClientFactory.getInstance()
+                CONNECTClient<AdapterPolicyEngineSecuredPortType> client = new CONNECTClientFactory<AdapterPolicyEngineSecuredPortType>()
                         .getCONNECTClientSecured(portDescriptor, url, assertion);
 
                 response = (CheckPolicyResponseType) client.invokePort(AdapterPolicyEngineSecuredPortType.class,
diff --git a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/proxy/PolicyEngineProxyWebServiceUnsecuredImpl.java b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/proxy/PolicyEngineProxyWebServiceUnsecuredImpl.java
index c2a2e58..a1b004f 100644
--- a/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/proxy/PolicyEngineProxyWebServiceUnsecuredImpl.java
+++ b/Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/policyengine/adapter/proxy/PolicyEngineProxyWebServiceUnsecuredImpl.java
@@ -31,7 +31,7 @@ import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
 import gov.hhs.fha.nhinc.common.nhinccommonadapter.CheckPolicyRequestType;
 import gov.hhs.fha.nhinc.common.nhinccommonadapter.CheckPolicyResponseType;
 import gov.hhs.fha.nhinc.messaging.client.CONNECTClient;
-import gov.hhs.fha.nhinc.messaging.client.CONNECTCXFClientFactory;
+import gov.hhs.fha.nhinc.messaging.client.CONNECTClientFactory;
 import gov.hhs.fha.nhinc.messaging.service.port.ServicePortDescriptor;
 import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
 import gov.hhs.fha.nhinc.nhinclib.NullChecker;
@@ -74,7 +74,7 @@ public class PolicyEngineProxyWebServiceUnsecuredImpl implements PolicyEnginePro
 
                 ServicePortDescriptor<AdapterPolicyEnginePortType> portDescriptor = new PolicyEngineUnsecuredServicePortDescriptor();
 
-                CONNECTClient<AdapterPolicyEnginePortType> client = CONNECTCXFClientFactory.getInstance()
+                CONNECTClient<AdapterPolicyEnginePortType> client = new CONNECTClientFactory<AdapterPolicyEnginePortType>()
                         .getCONNECTClientUnsecured(portDescriptor, url, assertion);
 
                 response = (CheckPolicyResponseType) client.invokePort(AdapterPolicyEnginePortType.class,
